### PR TITLE
radarr api testing

### PR DESCRIPTION
**Pull request by userdocs**
## Proposed Changes:
- Test for the api to be responsive instead of sleep and hope

## Tick these if they apply
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Some existing `utils` got changed

## Checklist
- [x] Docs have been made/are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [ ] I have commented my code, particularly in hard-to-understand areas

## Testing done
OS & Version:

### How have you tested this

Scenarios tested:
- I ran this and it worked
- I ran that and it didn't work

### CODE REVIEW

File: scripts/nginx/radarr.sh [hunk 1]
```diff
@@ -51,18 +51,19 @@ chown -R "$user":"$user" /home/"$user"/.config/Radarr
...
-	jq '.certificateValidation = "disabledForLocalAddresses"')"
+timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5;done'
+if [[ "$?" -ge "1" ]]; then
+	echo_warn "There is a problem talking to the Radarr API, we need to stop here and exit."
+	exit 1
+else
+	urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
+	echo_success "Radarr API tested and working!"
+fi
+
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:62]:
```suggestion
echo_progress_start "Waiting for Radarr API"
timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5;done'
if [[ "$?" -ge "1" ]]; then
	echo_warn "There is a problem talking to the Radarr API, we need to stop here and exit."
	exit 1
else
	urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
	echo_progress_done"Radarr API reachable"
fi
```

File: scripts/update/radarr.sh [hunk 2]
```diff
@@ -12,12 +12,19 @@ if [[ -f /install/.radarr.lock ]]; then
...
 		apikey=$(grep -oPm1 "(?<=<ApiKey>)[^<]+" /home/"${radarrOwner}"/.config/Radarr/config.xml)
 		echo_log_only "Apikey = $apikey"
-		# basicauth=$(echo "${radarrOwner}:$(_get_user_password ${radarrOwner})" | base64)
-		if [[ -f /install/.nginx.lock ]]; then
-			ret=$(curl -sS -L --insecure --user "${radarrOwner}":"$(_get_user_password "${radarrOwner}")" "http://127.0.0.1/radarr/api/v3/system/status?apiKey=${apikey}")
+		#
+		timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5;done'
+		#
+		if [[ "$?" -ge "1" ]]; then
+			echo_warn "There is a problem talking to the Radarr API, we need to stop here and exit."
```
Comment (flying-sausages) [inline scripts/update/radarr.sh:19]:
```suggestion
			echo_warn "Radarr API did not respond as expected. Please make sure Radarr is on v3 and running."
```

File: scripts/nginx/radarr.sh [hunk 3]
```diff
@@ -51,18 +51,19 @@ chown -R "$user":"$user" /home/"$user"/.config/Radarr
 . /etc/swizzin/sources/functions/utils
 systemctl start radarr -q # Switch radarr on regardless whether it was off before or not as we need to have it online to trigger this cahnge
 
-sleep 5 # TODO replace with a loop until the API is available
-payload="$(curl -s "https://127.0.0.1/radarr/api/v3/config/host?apiKey=${apikey}" \
-	--user "${user}:$(_get_user_password "${user}")" --insecure \
-	-s | \
-	jq '.certificateValidation = "disabledForLocalAddresses"')"
+timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5;done'
+if [[ "$?" -ge "1" ]]; then
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:55]:
```suggestion
if ! timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5; done' ; then
```

File: scripts/update/radarr.sh [hunk 4]
```diff
@@ -12,12 +12,19 @@ if [[ -f /install/.radarr.lock ]]; then
...
+			exit 1
 		else
-			ret=$(curl -sS -L --insecure "http://127.0.0.1:7878/api/v3/system/status?apiKey=${apikey}")
+			urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
+			echo_success "Radarr API tested and working!"
```
Comment (flying-sausages) [inline scripts/update/radarr.sh:23]:
```suggestion
			echo_log_only "Radarr API tested and working!"
```
or the other way around where we do start and done progress echos

**Review by flying-sausages — CHANGES_REQUESTED** [review]
Mostly consistencies with `echo` functions, and some little quickies. Still need to run this

File: scripts/nginx/radarr.sh [hunk 5]
```diff
@@ -51,18 +51,18 @@ chown -R "$user":"$user" /home/"$user"/.config/Radarr
...
-	--user "${user}:$(_get_user_password "${user}")" --insecure \
-	-s | \
-	jq '.certificateValidation = "disabledForLocalAddresses"')"
+echo_progress_start "Waiting for Radarr API"
+if ! timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5; done' ; then
+	echo_warn "There is a problem talking to the Radarr API, we need to stop here and exit."
+	exit 1
+else
+	urlbase="$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq '.urlBase' | cut -d '"' -f 2)"
+	echo_progress_done"Radarr API reachable"
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:60]:
```suggestion
	echo_progress_done "Radarr API reachable"
```

File: scripts/nginx/radarr.sh [hunk 6]
```diff
@@ -51,18 +51,18 @@ chown -R "$user":"$user" /home/"$user"/.config/Radarr
...
+fi
+payload=$(curl -sL "http://127.0.0.1:7878/api/v3/config/host?apikey=${apikey}" | jq ".certificateValidation = \"disabledForLocalAddresses\"")
 echo_log_only "Payload = \n${payload}"
 echo_log_only "Return from radarr after PUT ="
-curl "https://127.0.0.1/radarr/api/v3/config/host?apiKey=${apikey}" -X PUT --insecure \
-	-H 'Accept: application/json, text/javascript, */*; q=0.01' \
-	--compressed -H 'Content-Type: application/x-www-form-urlencoded; charset=UTF-8' \
-	--user "${user}:$(_get_user_password "${user}")" \
-	--data-raw "$payload" -s >> "$log"
+curl -so /dev/null "http://127.0.0.1:7878${urlbase}/api/v3/config/host?apikey=${apikey}" -X PUT -H 'Accept: application/json, text/javascript, */*; q=0.01' --compressed -H 'Content-Type: application/x-www-form-urlencoded; charset=UTF-8' --data-raw "${payload}" >> "$log"
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:65]: `-o /dev/null` would hide the return from the log, no?

Comment (userdocs) [reply scripts/nginx/radarr.sh:65]: it was to hide the output in my testing. it does not need to be there.

File: scripts/nginx/radarr.sh [hunk 7]
```diff
@@ -51,18 +51,20 @@ chown -R "$user":"$user" /home/"$user"/.config/Radarr
 . /etc/swizzin/sources/functions/utils
 systemctl start radarr -q # Switch radarr on regardless whether it was off before or not as we need to have it online to trigger this cahnge
 
-echo_progress_start "Waiting for Radarr API"
-if ! timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5; done' ; then
-	echo_warn "There is a problem talking to the Radarr API, we need to stop here and exit."
+echo_progress_start "Waiting for Radarr"
+if ! timeout 30 bash -c -- 'while ! curl -sfL "http://127.0.0.1:7878/api/v3/system/status?apiKey='"${apikey}"'" > /dev/null 2>&1; do sleep 5; done'; then
+	echo_warn "Radarr API did not respond as expected. Please make sure Radarr is on v3 and running."
```
Comment (flying-sausages) [inline scripts/nginx/radarr.sh:56]:
```suggestion
	echo_error "Radarr API did not respond as expected. Please make sure Radarr is on v3 and running."
```

**Pull request closed without merging**